extensions/amp-date-picker/1.0/component/date-range-picker.tsx

```diff
@@ -0,0 +1,432 @@
+      return;
+    }
+
+    const handleFocus = (event: FocusEvent) => {
+      if (event.target === startInputEl) {
+        setFocusedInput('start-input');
+      }
+      if (event.target === endInputEl) {
+        setFocusedInput('end-input');
+      }
+      transitionTo('overlay-open-input');
+    };
+
+    const handleDocumentKeydown = (event: KeyboardEvent) => {
+      if (event.key === Keys_Enum.ESCAPE) {
+        transitionTo('overlay-closed');
+      }
+    };
+    const handleInputKeydown = (event: KeyboardEvent) => {
+      const {target} = event;
+      if ((target as HTMLInputElement).type === 'hidden') {
+        return;
+      }
+
+      if (event.key === Keys_Enum.DOWN_ARROW) {
+        transitionTo('overlay-open-picker');
+      }
+    };
+    const handleClick = (event: MouseEvent) => {
+      // This is necessary for bento mode, since .contains does not work with
+      // elements in the shadow DOM
+      const containsShadowRoot = !!(event.target as Element)?.shadowRoot;
+      const clickWasInDatePicker =
+        event.target === startInputEl ||
+        event.target === endInputEl ||
+        containerEl.contains(event.target as Node) ||
+        containsShadowRoot;
+      if (!clickWasInDatePicker) {
+        transitionTo('overlay-closed');
+      }
+    };
+    if (mode === 'overlay') {
+      document.addEventListener('click', handleClick);
+      document.addEventListener('keydown', handleDocumentKeydown);
+    }
+    startInputEl?.addEventListener('focus', handleFocus);
+    endInputEl?.addEventListener('focus', handleFocus);
+    startInputEl?.addEventListener('change', handleStartInput);
+    endInputEl?.addEventListener('change', handleEndInput);
+    startInputEl?.addEventListener('keydown', handleInputKeydown);
+    endInputEl?.addEventListener('keydown', handleInputKeydown);
+    return () => {
+      document.removeEventListener('click', handleClick);
+      document.removeEventListener('keydown', handleDocumentKeydown);
+      startInputEl?.removeEventListener('focus', handleFocus);
+      endInputEl?.removeEventListener('focus', handleFocus);
+      startInputEl?.removeEventListener('change', handleStartInput);
+      endInputEl?.removeEventListener('change', handleEndInput);
+      startInputEl?.removeEventListener('keydown', handleInputKeydown);
+      endInputEl?.removeEventListener('keydown', handleInputKeydown);
+    };
+  }, [
+    transitionTo,
+    mode,
+    handleStartInput,
+    handleEndInput,
+    startDateInput.ref,
+    endDateInput.ref,
+  ]);
+
+  const disabledMatchers = useMemo(() => {
+    const matchers: Matcher[] = [isDisabled];
+    if (startDateInput.date) {
+      const disabledAfter = getDisabledAfter(startDateInput.date);
+      if (disabledAfter) {
+        matchers.push({after: disabledAfter});
+      }
+    }
+    return matchers;
+  }, [getDisabledAfter, isDisabled, startDateInput.date]);
+
+  return (
+    <ContainWrapper
+      class="amp-date-picker-calendar-container"
```

Comment:
I'm not 100% sure if this class does anything, since it's inside the Bento component / shadow ODM.
From my understanding, the `jss` file applies to the shadow DOM, and the `.css` file applies to the Light DOM.

Reply:
Hmm, interesting! At the moment I'm using it to find the calendar component in the e2e tests, but I am having issues with CSS in the shadow DOM. React day picker has an imported CSS file from the library, but I'm not sure how to convert that to JSS or apply those styles to the shadow DOM.